etna/datasets/tsdataset.py

```diff
@@ -416,10 +417,26 @@ def inverse_transform(self, transforms: Sequence["Transform"]):
         Applied in reversed order.
         """
         # TODO: return regressors after inverse_transform
+        inverse_transform_target_components_flg = self.target_components is not None
+        target_df, target_components_df = None, None
+        if inverse_transform_target_components_flg:
+            target_df = self.to_pandas(features=["target"])
+            target_components_df = self.get_target_components()
+            self.drop_target_components()
+
         for transform in reversed(transforms):
             tslogger.log(f"Inverse transform {repr(transform)} is applied to dataset")
             transform.inverse_transform(self)
```

Comment:
Can you explain, why do we need this logic in `TSDataset` if changing of components is already made in `transform.inverse_transform`?

Reply:
To prevent several inverse transformation of target components

Reply:
Ok, understood